### PR TITLE
Integration

### PR DESCRIPTION
**Pull request by Fahim-Khan-P**
### **We added the following:**

- [ ] Before N+1 Problem:

- [ ] After Fixing N+1:

- Use Capybara to write integration tests for each view in your project.

- [ ] Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- [ ] User index page:
1. I can see the username of all other users.
2. I can see the profile picture for each user.
3. I can see the number of posts each user has written.
4. When I click on a user, I am redirected to that user's show page.

- [ ] user show page:
1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see the user's bio.
5. I can see the user's first 3 posts.
6. I can see a button that lets me view all of a user's posts.
7. When I click a user's post, it redirects me to that post's show page.
8. When I click to see all posts, it redirects me to the user's post's index page.

- [ ] User post index page:
1. I can see the user's profile picture.
2. I can see the user's username.
3. I can see the number of posts the user has written.
4. I can see a post's title.
5. I can see some of the post's body.
6. I can see the first comments on a post.
7. I can see how many comments a post has.
8. I can see how many likes a post has.
9. I can see a section for pagination if there are more posts than fit on the view.
10. When I click on a post, it redirects me to that post's show page.

- [ ] Post show page:
1. I can see the post's title.
2. I can see who wrote the post.
3. I can see how many comments it has.
4. I can see how many likes it has.
5. I can see the post body.
6. I can see the username of each commentor.
7. I can see the comment each commentor left.

### CODE REVIEW

File: spec/feature/users_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        expect(page).to have_content(@third_post.text)
+      end
+      it "should have a button to view all of a user's posts" do
+        expect(page).to have_link('See all posts', href: user_posts_path(user_id: @user.id))
+      end
+      it 'should redirects to all posts show page.' do
+        click_link 'See all posts'
+        expect(page).to have_current_path(user_posts_path(user_id: @user.id))
+      end
+    end
```
Comment (TedLivist) [inline spec/feature/users_show_spec.rb:78]: - [ ] Kindly add tests that assert the following:

  - Tests that asserts you can see the user's profile picture.
  - Tests that redirects to the post's show page when you click on a user's post.

File: spec/feature/posts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      expect(page).to have_content(@post.likes_counter)
+    end
+    it 'displays the post author name' do
+      expect(page).to have_content(@post.user.name)
+    end
+    it 'redirects to posts show page' do
+      click_link @post4.title
+      expect(page).to have_current_path user_post_path(@user, @post4)
+    end
+  end
```
Comment (TedLivist) [inline spec/feature/posts_spec.rb:54]: - [ ] Kindly add a test for the following scenario.

  - Tests that asserts you can see some of the post's body.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @Fahim-Khan-P, @Thinus01 and @lawrahkonwea,

Good job so far!
✔️ Tests are well-written and passing
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/feature/users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      expect(page).to have_content(@user.name)
+      expect(page).to have_content(@user2.name)
+      expect(page).to have_content(@user3.name)
+      expect(page).to have_content(@user4.name)
+      expect(page).to have_content(@user5.name)
+    end
+    it 'dispalys user image' do
+      visit users_path
+      expect(page).to have_css("img[src='https://unsplash.com/photos/WNoLnJo7tS8']")
+    end
```
Comment (codepantha) [inline spec/feature/users_spec.rb:29]: - [ ] This test is failing as you can see in the attached image below. Please try to make it pass
image

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**